src/overlays/kaleido_scope/ovl_kaleido_scope/z_kaleido_map.c

```diff
@@ -451,28 +457,32 @@ void KaleidoScope_UpdateDungeonCursor(PlayState* play) {
 }
 
 TexturePtr sCloudTextures[] = {
-    gWorldMapCloud1Tex,  // TINGLE_MAP_CLOCK_TOWN
-    gWorldMapCloud2Tex,  // TINGLE_MAP_CLOCK_TOWN
-    gWorldMapCloud13Tex, // TINGLE_MAP_WOODFALL
-    gWorldMapCloud14Tex, // TINGLE_MAP_WOODFALL
-    gWorldMapCloud15Tex, // TINGLE_MAP_WOODFALL
-    gWorldMapCloud8Tex,  // TINGLE_MAP_SNOWHEAD
-    gWorldMapCloud9Tex,  // TINGLE_MAP_SNOWHEAD
-    gWorldMapCloud10Tex, // TINGLE_MAP_SNOWHEAD
-    gWorldMapCloud3Tex,  // TINGLE_MAP_ROMANI_RANCH
-    gWorldMapCloud4Tex,  // TINGLE_MAP_GREAT_BAY
-    gWorldMapCloud5Tex,  // TINGLE_MAP_GREAT_BAY
-    gWorldMapCloud6Tex,  // TINGLE_MAP_GREAT_BAY
-    gWorldMapCloud7Tex,  // TINGLE_MAP_GREAT_BAY
-    gWorldMapCloud11Tex, // TINGLE_MAP_STONE_TOWER
-    gWorldMapCloud12Tex, // TINGLE_MAP_STONE_TOWER
+    gWorldMapClockTownCloud1Tex,  // TINGLE_MAP_CLOCK_TOWN
+    gWorldMapClockTownCloud2Tex,  // TINGLE_MAP_CLOCK_TOWN
+    gWorldMapWoodfallCloud1Tex,   // TINGLE_MAP_WOODFALL
+    gWorldMapWoodfallCloud2Tex,   // TINGLE_MAP_WOODFALL
+    gWorldMapWoodfallCloud3Tex,   // TINGLE_MAP_WOODFALL
+    gWorldMapSnowheadCloud1Tex,   // TINGLE_MAP_SNOWHEAD
+    gWorldMapSnowheadCloud2Tex,   // TINGLE_MAP_SNOWHEAD
+    gWorldMapSnowheadCloud3Tex,   // TINGLE_MAP_SNOWHEAD
+    gWorldMapRomaniRanchCloudTex, // TINGLE_MAP_ROMANI_RANCH
+    gWorldMapGreatBayCloud1Tex,   // TINGLE_MAP_GREAT_BAY
+    gWorldMapGreatBayCloud2Tex,   // TINGLE_MAP_GREAT_BAY
+    gWorldMapGreatBayCloud3Tex,   // TINGLE_MAP_GREAT_BAY
+    gWorldMapGreatBayCloud4Tex,   // TINGLE_MAP_GREAT_BAY
+    gWorldMapStoneTowerCloud1Tex, // TINGLE_MAP_STONE_TOWER
+    gWorldMapStoneTowerCloud2Tex, // TINGLE_MAP_STONE_TOWER
 };
-s16 D_8082B584[] = {
-    0, 0, 255, 255, 255, 0,
+
+s16 sWorldMapDotPrimColors[][3] = {
+    { 0, 0, 255 },
+    { 255, 255, 0 },
 };
-s16 D_8082B590[] = {
-    255, 255, 0, 0, 0, 255,
+s16 sWorldMapDotEnvColors[][3] = {
+    { 255, 255, 0 },
+    { 0, 0, 255 },
 };
```

Comment:
These are only ever used as 2D arrays, right?

Reply:
It's hard to say for sure with this particular example since the second half goes unused, but it's almost always broken up into a 2D array like this instead of a single r-b-g-r-b-g value. Maybe it's an unused pulsating target. But this seems much more likely than what it was before

Reply:
What would cause UB here?

Reply:
Accessing something declared as a 2D array as if it were a 1D array and indexing oob of the first subarray is technically UB.

Reply:
Right, I've never looked at those cases in player, but here, indexing should be limited to [0] -> red, [1] -> green, [2] -> blue, so you shouldn't ever be indexing it as a 1D array to begin with I think?

Reply:
Yeah, it's moot if the second subarray isn't accessed.

Reply:
Tbh many of these 2D color array structs that are all over the interface devs work could probably be made into an 1D array s16 rgb struct